src/sdk.spec.js

```diff
@@ -91,70 +48,69 @@ describe('OpenForm', () => {
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       languageSelectorTarget: '#my-languages-element',
       lang: 'nl',
     });
 
-    try {
-      await act(async () => await form.init());
-    } catch (e) {
-      throw e; // should not error
-    }
+    await act(async () => await form.init());
 
+    // wait for the loader to be removed when all network requests have completed
+    await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
     expect(target).not.toBeEmptyDOMElement();
   });
 
   it('should render the form on init', async () => {
     const formRoot = document.createElement('div');
-
-    apiModule.get.mockImplementation(mockAPICalls);
-
+    mswServer.use(...apiMocks);
     const form = new OpenForm(formRoot, {
-      baseUrl: 'http://localhost:8000/api/v2/',
+      baseUrl: BASE_URL,
       basePath: '',
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       lang: 'nl',
     });
 
-    try {
-      await act(async () => await form.init());
-    } catch (e) {
-      throw e; // should not error
-    }
+    await act(async () => await form.init());
 
+    // wait for the loader to be removed when all network requests have completed
+    await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
     expect(formRoot.textContent).not.toContain('Loading');
   });
 
   it('should re-fetch the form to get new literals after language change', async () => {
     const formRoot = document.createElement('div');
-
-    apiModule.get.mockImplementation(mockAPICalls);
+    // first we load NL variant, second time we load the form in NL
+    const formNL = getForm({translationEnabled: true, name: 'Nederlandse versie'});
+    const formEN = getForm({translationEnabled: true, name: 'English version'});
+    mswServer.use(
+      mockFormGet(formNL, true),
+      mockFormGet(formEN, true),
+      mockLanguageInfoGet(LANGUAGES),
+      mockFormioTranslations
+    );
 
     const form = new OpenForm(formRoot, {
-      baseUrl: 'http://localhost:8000/api/v2/',
+      baseUrl: BASE_URL,
       basePath: '',
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       lang: 'nl',
     });
 
-    try {
-      await act(async () => {
-        await form.init();
-        await form.onLanguageChangeDone('en');
-      });
-    } catch (e) {
-      throw e; // should not error
-    }
-
-    const is_get_form_call = args => args.length == 1 && args[0] == FORM.url;
-    const get_form_calls = apiModule.get.mock.calls.filter(is_get_form_call);
-    expect(get_form_calls.length).toBe(2);
+    await act(async () => {
+      await form.init();
+      await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
+    });
+
+    expect(within(formRoot).getAllByText('Nederlandse versie').length).toBeGreaterThan(0);
+
+    await act(async () => {
+      await form.onLanguageChangeDone('en');
+    });
+
+    expect(within(formRoot).getAllByText('English version').length).toBeGreaterThan(0);
```

Comment:
I've take a different approach here - rather than asserting that X number of API calls were made, the expected result is asserted. First load the form in NL -> Dutch form title visible, then change the language and this should result in the English variants being displayed instead.

Reply:
Nice,  with the new mock the api is abstracted away. How these strings appear is not important.